cloudinit/sources/DataSourceVMware.py

```diff
@@ -821,38 +821,73 @@ def is_valid_ip_addr(val):
     )
 
 
-def convert_to_netifaces_format(addr):
+def convert_to_netifaces_ipv4_format(addr):
     """
     Takes a cloudinit.netinfo formatted address and converts to netifaces
     format, since this module was originally written with netifaces as the
     network introspection module.
-    netifaces format:
+    netifaces ipv4 format:
     {
       "broadcast": "10.15.255.255",
       "netmask": "255.240.0.0",
       "addr": "10.0.1.4"
     }
-
-    cloudinit.netinfo format:
+    cloudinit.netinfo ipv4 format:
     {
       "ip": "10.0.1.4",
       "mask": "255.240.0.0",
       "bcast": "10.15.255.255",
       "scope": "global",
     }
     """
-    return {
-        "broadcast": addr["bcast"],
-        "netmask": addr["mask"],
-        "addr": addr["ip"],
+    addr_in_netifaces_format = {}
+    if "bcast" in addr:
+        addr_in_netifaces_format["broadcast"] = addr["bcast"]
+    if "mask" in addr:
+        addr_in_netifaces_format["netmask"] = addr["mask"]
+    if "ip" in addr:
+        addr_in_netifaces_format["addr"] = addr["ip"]
+    return addr_in_netifaces_format
+
+
+def convert_to_netifaces_ipv6_format(addr):
+    """
+    Takes a cloudinit.netinfo formatted address and converts to netifaces
+    format, since this module was originally written with netifaces as the
+    network introspection module.
+    netifaces ipv6 format:
+    {
+      "netmask": "ffff:ffff:ffff:ffff::/64",
+      "addr": "2001:db8:abcd:1234::1"
+    }
+    cloudinit.netinfo ipv6 format:
+    {
+      "ip": "2001:db8:abcd:1234::1/64",
+      "scope6": "global",
     }
+    """
+    addr_in_netifaces_format = {}
```

Comment:
This function does manual string parsing, which shouldn't be required due to Python standard library code.

Maybe you could use something like this? (I haven't tested it)

```python
def convert_to_netifaces_ipv6_format(addr: dict) -> dict:
    """
    Takes a cloudinit.netinfo formatted address and converts to netifaces
    format, since this module was originally written with netifaces as the
    network introspection module.
    netifaces ipv6 format:
    {
      "netmask": "ffff:ffff:ffff:ffff::/64",
      "addr": "2001:db8:abcd:1234::1"
    }
    cloudinit.netinfo ipv6 format:
    {
      "ip": "2001:db8:abcd:1234::1/64",
      "scope6": "global",
    }
    """
    if not addr.get("ip"):
        return {}
    ipv6 = ipaddress.IPv6Interface(addr.get("ip"))
    return {
        "netmask": f"{ipv6.netmask}/{ipv6.network.prefixlen}",
        "addr": str(ipv6.ip)
    }
```

Reply:
Updated this function and the above convert_to_netifaces_ipv4_format function as suggested, thanks!